util/genrest/resttools/constants.go

```diff
@@ -56,3 +56,6 @@ func init() {
 
 	RegexLiteral = fmt.Sprintf(`^[%s]+`, CharsLiteral)
 }
+
+// A key-type for storing binding URI value in the Context
+type BindingURIKey string
```

Comment:
all the uses of this are `BindingUriKey("bindingUri")`, AFICT. Why not define `const BindingUriKey string="bindingUri"`?

Reply:
:+1:

Reply:
the constant has it's own string-derived type instead of string because (as I understand) the Context is a shared space so if someone else wants to have a "bindingUri" it'll be in a separate type.

Reply:
Oh, I see what you mean about the type. The [doc](https://pkg.go.dev/context#WithValue) stipulates using non-built-in types to avoid collisions with other packages, since the type, not just the name,  presumably plays a part in the lookup.